tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -1455,4 +1455,108 @@ TEST(RDFSnapshotMore, ZeroOutputEntriesMT)
    gSystem->Unlink(fname);
 }
 
+// Test for custom basket size in Snapshot
+class SnapshotCustomBasketRAII {
+private:
+   std::string fInputFile;
+   std::string fOutputFileCustom;
+   std::string fOutputFileCollection;
+   std::unique_ptr<TFile> fFile;
+   std::unique_ptr<TTree> fTree;
+
+public:
+   SnapshotCustomBasketRAII() : 
+      fInputFile("input_file.root"),
+      fOutputFileCustom("output_file_custom_basket.root"),
+      fOutputFileCollection("output_file_collection_basket.root")
+   {
+      // Create input file and tree
+      fFile.reset(TFile::Open(fInputFile.c_str(), "RECREATE"));
+      fTree.reset(new TTree("tree", "Test Tree"));
```

Comment:
No need to use `unique_ptr` in this case. Just create the TFile and TTree on the stack in the constructor body.